components/inputs/input-number.js

```diff
@@ -78,13 +114,20 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 				this._formattedValue = '';
 				this._value = undefined;
 			} else {
-				this._formattedValue = formatValue(val, this.minFractionDigits, this.maxFractionDigits);
+				this._formattedValue = formatValue(
+					val,
+					this.minFractionDigits,
+					this.maxFractionDigits,
+					this.trailingZeroes,
+					this._numDecimalDigits
+				);
 				this._value = parseNumber(this._formattedValue);
 			}
 		} catch (err) {
 			this._formattedValue = '';
 			this._value = undefined;
 		}
+		this._numDecimalDigits = 0;
```

Comment:
We need to reset this so that subsequent calls to set the `value` property don't magically sprout the previous value's trailing zeroes count.